python/cog/server/worker.py

```diff
@@ -248,25 +281,24 @@ def _consume_events_inner(self) -> None:
                 self._result = None
             self._state = WorkerState.DEFUNCT
 
-    def _publish(self, ev: _PublicEventType) -> None:
-        for subscriber in self._subscribers.values():
-            try:
-                subscriber(ev)
-            except Exception:
-                log.warn("publish failed", subscriber=subscriber, ev=ev, exc_info=True)
-
-
-class LockedConn:
-    def __init__(self, conn: Connection) -> None:
-        self.conn = conn
-        self._lock = _spawn.Lock()
-
-    def send(self, obj: Any) -> None:
-        with self._lock:
-            self.conn.send(obj)
-
-    def recv(self) -> Any:
-        return self.conn.recv()
+    def _publish(self, e: Envelope) -> None:
+        log.warn(f"handling {e.event} for prediction {e.tag}")
+        for subscriber, requested_tag in self._subscribers.values():
+            if requested_tag is None or e.tag == requested_tag:
+                try:
+                    subscriber(cast(_PublicEventType, e.event))
+                except Exception:
+                    log.warn(
+                        "publish failed",
+                        subscriber=subscriber,
+                        tag=e.tag,
+                        event=e.event,
+                        exc_info=True,
+                    )
+            else:
+                log.warn(
+                    f"skipping publish for {e.event}, because {requested_tag} is None or {e.tag} != {requested_tag}"
+                )
```

Comment:
```suggestion
            else:
                log.warn(
                    f"skipping publish for {e.event}, because {requested_tag} is None or {e.tag} != {requested_tag}"
                )
```

This looks like leftover debugging we probably don't want to ship.